src/Core/Configurations/RuntimeConfigValidator.cs

```diff
@@ -225,11 +331,15 @@ public async Task ValidateEntitiesMetadata(RuntimeConfig runtimeConfig, ILoggerF
             logger: loggerFactory.CreateLogger<ISqlMetadataProvider>(),
             fileSystem: _fileSystem,
             isValidateOnly: _isValidateOnly);
-
         await metadataProviderFactory.InitializeAsync();
+
         ConfigValidationExceptions.AddRange(metadataProviderFactory.GetAllMetadataExceptions());
 
-        ValidateRelationshipsInConfig(runtimeConfig, metadataProviderFactory);
+        // Validation below relies on metadata being populated from backend, only execute when there were no connection errors
+        if (!ConfigValidationExceptions.Any(x => x.Message.StartsWith(DataApiBuilderException.CONNECTION_STRING_ERROR_MESSAGE)))
+        {
+            ValidateEntityRelationshipsInConfigWithProvidedMetadata(runtimeConfig, metadataProviderFactory);
```

Comment:
The method names seems redundant:
- ValidateEntityRelationshipsInConfigWithGeneratedMetadata
- ValidateEntityRelationshipsInConfigWithProvidedMetadata
- ValidateEntityRelationshipsInConfigWithoutDBMetadata

Some of the methods say "Metadata" while one says "DbMetadata" and then you also have "Generated" and "Provided" metadata. Naming needs to be updated with comments to clarify what the difference is. The first two seem to be the same.

Reply:
That was the intention of the summaries

* ValidateEntityRelationshipsInConfigWithGeneratedMetadata
    /// This method validates the entities relationships against the database objects using
    /// metadata from the backend DB generated by this function.
    
* ValidateEntityRelationshipsInConfigWithProvidedMetadata
    /// Validates the semantic correctness of an Entity's relationships in the
    /// runtime configuration against the metadata collected from the backend DB
    /// that is provided to this function via use of the MetadataProviderFactory

* ValidateEntityRelationshipsInConfigWithoutDBMetadata
    /// Validates the semantic correctness of an Entity's relationships in the
    /// runtime configuration when metadata from the backend DB is not required.


To summarize, the "generated" function will generate the metadata by creating a meta data factory and then initializing a metadata provider. This will go to the DB and populate the provider with metadata, that provider is then used to call the provided function.

the reason we have the matadata-generated function calling the metadata-provided function is that we already have code that is trying to validate using their own meta data provider, so we should keep that as an option.

I can add more details if you think its worth it.

Reply:
It's a bit confusing with those names being so verbose: `ValidateEntityRelationshipsInConfig...` where both "generated" and "provided" are public methods.
- (public) `ValidateEntityRelationshipsInConfigWithoutDBMetadata()`
- (public) `ValidateEntityRelationshipsInConfigWithGeneratedMetadataAsync()`
  - (public*) `ValidateEntityRelationshipsInConfigWithProvidedMetadata()`
    - (private) `ValidateSourceAndTargetFieldsAsBackingColumns()`

* Could all functions listed in sub-bullets of `ValidateEntityRelationshipsInConfigWithGeneratedMetadataAsync()` be private? The top level validation function can be generic: ValidateEntityRelationshipsAgainstDb()
and then whatever helper functions get called internally to that function can be named to more specifically denote what they're validating.

Reply:
We synced offline, but just to give context here. Since the metadata provider initialization is handling validation, we have the outer call which handles validation of entity relationships also handling validation of the entities, and is named `ValidateEntitiesMetadata`, this function then calls into a function that only validated the relationship information that requires metadata to proceed, `ValidateRelationships`, and the relationship validation that does not need metadata happens on its own and is named `ValidateRelationshipConfigCorrectness`.

`ValidateEntitiesMetadata` is always called during dab validate, as is `ValidateRelationshipConfigCorrectness`, however `ValidateRelationships` is only called from withing `ValidateEntitiesMetadata` when there are no connection errors during the initialization of the metadata provider.